app/src/main/java/com/termux/api/apis/BluetoothScanAPI.java

```diff
@@ -0,0 +1,122 @@
+package com.termux.api.apis;
+
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothDevice;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.util.JsonWriter;
+
+import com.termux.api.util.ResultReturner;
+import com.termux.shared.logger.Logger;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BluetoothScanAPI {
+
+    private static final ArrayList<BluetoothDevice> devices = new ArrayList<>();
+    private static final String LOG_TAG = "BluetoothScanAPI";
+    private static BluetoothAdapter adapter;
+    private static boolean scanning = false;
+
+    private static final BroadcastReceiver receiver = new BroadcastReceiver() {
```

Comment:
Please give this a better name, even though it's the only `BroadcastReceiver` in sight. `device_found_receiver`?